sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -669,6 +751,11 @@ static int sof_audio_probe(struct platform_device *pdev)
 
 	sof_audio_card_rt5682.dai_link = dai_links;
 
+	if (!strcmp(pdev->name, "jsl_rt1015_rt5682")) {
```

Comment:
@yongzhi1 do you think it makes sense to add a quirk for the AMP type rt1015/max for this change instead of using the driver name? It will make reusing the machine driver for a different platform with the same codec combination.

Reply:
Yes, the driver is for Google waddle-dee, but the DMI data(product name etc.) is not yet finalized to use quirk cb.

Reply:
but you could also do the same in the driver data no?

Reply:
yes, you are right, I can define a new BIT for the rt1015, and use driver_data to set it.